### PR TITLE
Remove leftover `@ApiStatus.Internal` annotations and add annotation validation

### LINKED ISSUE #2822: Remove leftover `@ApiStatus.Internal` annotations
Since the annotations are now autogenerated for impl classes, most of the remaining annotations can be removed.

### PR DESCRIPTION
**Pull request by Technici4n**
Fixes #2822.

### CODE REVIEW

File: gradle/validate-annotations.gradle [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+import static groovy.io.FileType.FILES
+
+class ValidateAnnotations extends SourceTask {
+	private static final def API_STATUS_INTERNAL = ~/@ApiStatus\.Internal/
+	private static final def ENVIRONMENT = ~/@Environment/
+
+	@SkipWhenEmpty
+	@InputDirectory
+	final DirectoryProperty sourceRoot = project.objects.directoryProperty()
```
Comment (modmuss50) [inline gradle/validate-annotations.gradle]: This isnt needed when using `SourceTask`. 

You can use getSource https://github.com/gradle/gradle/blob/master/subprojects/core/src/main/java/org/gradle/api/tasks/SourceTask.java#L58

Comment (Technici4n) [reply gradle/validate-annotations.gradle]: Thanks. How about now?

File: gradle/validate-annotations.gradle [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+	// We have to capture the source set name for the lazy string literals,
+	// otherwise it'll just be whatever the last source set is in the list.
+	def sourceSetName = sourceSet.name
+	def taskName = sourceSet.getTaskName('validate', 'Annotations')
+	def task = tasks.register(taskName, ValidateAnnotations) {
+		group = 'fabric'
+		description = "Validate annotations for $sourceSetName packages."
+
+		// Only apply to default source directory since there's also generated package-info files.
+		setSource file("src/$sourceSetName/java")
```
Comment (modmuss50) [inline gradle/validate-annotations.gradle]: Looking better, can this become a single task with the following as the input?:

```gradle
source sourceSets.main.allSource
source sourceSets.client.allSource
```

Comment (Technici4n) [reply gradle/validate-annotations.gradle]: Wouldn't that include the generated package-info files?

Comment (modmuss50) [reply gradle/validate-annotations.gradle]: Ah yeah, it would do 👍 

I still think you should be able to make this one task though.

Comment (Technici4n) [reply gradle/validate-annotations.gradle]: How about now? :smile:

Comment (modmuss50) [reply gradle/validate-annotations.gradle]: Looks great, thanks 👍

File: gradle/validate-annotations.gradle [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+				}
+
+				def contents = it.text
+
+				// @Environment is never allowed
+				if (ENVIRONMENT.matcher(contents).find()) {
+					throw new RuntimeException("Found @Environment annotation in file: $it")
+				}
+
+				// @ApiStatus.Internal is only allowed in api packages (it's auto-generated for impl packages)
```
Comment (Juuxel) [inline gradle/validate-annotations.gradle]:
```suggestion
				// @ApiStatus.Internal is only allowed in api packages (it's auto-generated for impl and mixin packages)
```

**Review by Juuxel — APPROVED** [review]

**Review by modmuss50 — APPROVED** [review]

**Pull request merged**